frinx/workers/test/test_worker.py

```diff
@@ -0,0 +1,169 @@
+import random
+
+from frinx.common.conductor_enums import TaskResultStatus
+from frinx.common.worker.service import ServiceWorkersImpl
+from frinx.common.worker.task_def import TaskDefinition
+from frinx.common.worker.task_def import TaskInput
+from frinx.common.worker.task_def import TaskOutput
+from frinx.common.worker.task_result import TaskResult
+from frinx.common.worker.worker import WorkerImpl
+
+
+class TestWorker(ServiceWorkersImpl):
+    class Echo(WorkerImpl):
+        class WorkerDefinition(TaskDefinition):
+            name = 'TEST_echo'
+            description = 'testing purposes: returns input unchanged'
+            labels = ['TEST']
+            timeout_seconds = 60
+            response_timeout_seconds = 60
+
+        class WorkerInput(TaskInput):
+            input: str
+
+        class WorkerOutput(TaskOutput):
+            output: str
+
+        def execute(self, worker_input: WorkerInput) -> TaskResult:
+            task_result = TaskResult(
+                status=TaskResultStatus.COMPLETED, logs=['Echo worker invoked successfully']
+            )
+            task_result.add_output_data('output', worker_input.input)
+            return task_result
+
+    ###############################################################################
+
+    class Sleep(WorkerImpl):
+        DEFAULT_SLEEP = 10
+        MAX_SLEEP = 600
+
+        class WorkerDefinition(TaskDefinition):
+            name = 'TEST_sleep'
+            description = 'testing purposes: sleep'
+            labels = ['TEST']
+            timeout_seconds = 600
+            response_timeout_seconds = 600
+
+        class WorkerInput(TaskInput):
+            time: int | None = 10
+
+        class WorkerOutput(TaskOutput):
+            time: int
+
+        def execute(self, worker_input: WorkerInput) -> TaskResult:
+            sleep = worker_input.time if worker_input.time else self.DEFAULT_SLEEP
+            if sleep < 0 or sleep > self.MAX_SLEEP:
+                return TaskResult(
+                    status=TaskResultStatus.FAILED,
+                    logs=['Invalid sleep time, must be > 0 and < 600'],
+                )
+
+            import time
+
+            task_result = TaskResult(
+                status=TaskResultStatus.COMPLETED, logs=['Sleep worker invoked. Sleeping']
+            )
+            time.sleep(sleep)
+            task_result.add_output_data('time', sleep)
+            return task_result
```

Comment:
Just a suggestion. I find this style more readable.

```suggestion
            time.sleep(sleep)
            return TaskResult(
                status=TaskResultStatus.COMPLETED,
                logs=['Sleep worker invoked. Sleeping'],
                output={'time': sleep}
            )
```